Tests/ImageFetcherTests/Helpers.swift

```diff
@@ -99,3 +99,67 @@ extension URLSessionConfiguration {
         return config
     }
 }
+
+class MockCache: Cache {
+    struct CacheError: Error {
+        let reason: String
+    }
+
+    var onCache: (Data, String) throws -> ()
+    var onData: (String) throws -> Data
+    var onDelete: (String) throws -> ()
+    var onDeleteAll: () throws -> ()
+    var onFileURL: (String) -> URL
+
+    init(
+        onCache: @escaping (Data, String) throws -> () = { _, _ in },
+        onData: @escaping (String) throws -> Data = { _ in Data() },
+        onDelete: @escaping (String) throws -> () = { _ in },
+        onDeleteAll: @escaping () throws -> () = { },
+        onFileURL: @escaping (String) -> URL = { _ in URL(fileURLWithPath: "") }
```

Comment:
I considered calling `XCTFail` in by default but that ended up complicating this design if I wanted the failure to indicate the method call responsible